### PR TITLE
Allow to customize JdbcIdentityCacheMapping for CredentialProvider

### PR DESCRIPTION
**Pull request by kokosing**
Allow to customize JdbcIdentityCacheMapping for CredentialProvider

### CODE REVIEW

**Comment by kokosing** [comment]
Relates to https://github.com/trinodb/trino/pull/7039

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPlugin.java [hunk 1]
```diff
@@ -40,6 +40,11 @@ public JdbcPlugin(String name, Module module)
     @Override
     public Iterable<ConnectorFactory> getConnectorFactories()
     {
-        return ImmutableList.of(new JdbcConnectorFactory(name, combine(new CredentialProviderModule(), module)));
+        return ImmutableList.of(new JdbcConnectorFactory(
+                name,
+                combine(
+                        new CredentialProviderModule(),
+                        new ExtraCredentialsBasedJdbcIdentityCacheMappingModule(),
```
Comment (hashhar) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPlugin.java:47]: Is this binding "needed" for every connector? Or should each connector module be responsible to install this module themselves?

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ExtraCredentialsBasedJdbcIdentityCacheMappingModule.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+
+class ExtraCredentialsBasedJdbcIdentityCacheMappingModule
```
Comment (findepi) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ExtraCredentialsBasedJdbcIdentityCacheMappingModule.java:20]: public

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPlugin.java (hunk 1 above)
Comment (findepi) [reply plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPlugin.java:47]: This is an OK change from refactor perspective, as long as all `CredentialProviderModule` usages get updated, but you're right, we could want a different refactor instead, giving each connector more control, and then we could subsequently remove unneeded occurrences.

**Review by findepi — APPROVED** [review]
% @hashhar 's cmt

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPlugin.java (hunk 1 above)
Comment (kokosing) [reply plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPlugin.java:47]: > Is this binding "needed" for every connector? 

It is as this is required for `CachingJdbcClient`, however it is not always a good thing to install this module together with `CredentialProviderModule`

**Pull request merged**